api/src/test/scala/dcos/metronome/api/v1/controllers/JobSpecControllerTest.scala

```diff
@@ -285,16 +285,15 @@ class JobSpecControllerTest extends PlaySpec with OneAppPerTestWithComponents[Mo
       contentType(response) mustBe Some("application/json")
     }
 
-    "indicate a problem when creating a job without a secret name" in {
+    "indicate no problem when creating a job with unused secrets" in {
```

Comment:
it took me time to figure out why we needed this change but 👍 

Btw. it is a bit controversial because having extra secret might mean that someone forgot to actually use it which he/she will figure out too late, in the runtime...

Reply:
I've changed it (and the JIRA description) to fail in both scenarios, with a meaningful message.